### PR TITLE
Fixes #792 cmd empty echo

### LINKED ISSUE #792: cmd handles empty echo incorrectly
With `cmd` any `info('')` call results in `echo` which in `cmd` shows the status of the echo.
To print an empty line `echo.` is needed instead.

Reported by @instinct-vfx, introduced at:
https://github.com/nerdvegas/rez/commit/b277d26d8d76f28c0e05b0a60158b5d03009c96d#diff-b1aa7ab463cecaa539089580fe418dc2R164

### PR DESCRIPTION
**Pull request by bfloch**
Without it `info('')` generates `echo is ON.` on cmd's stdout.

EDIT: More context
In `cmd` and `info('')` results in `echo ""` which is interpreted as command to show the current state of echo.
The way to echo an empty line is `echo.`.

I've also added a test that makes sure that info('') is interpreted as empty line.

### CODE REVIEW

File: src/rezplugins/shell/cmd.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def get_syspaths(cls):
 
         # detect system paths using registry
         def gen_expected_regex(parts):
-            whitespace = "[\s]+"
+            whitespace = r"[\s]+"
```
Comment (bfloch) [inline src/rezplugins/shell/cmd.py]: Eliminates a warning (invalid escape sequence)

**Comment by bfloch** [comment]
I reran the tests since we did not have the PR workflow fixed back then.
Ready for review @nerdvegas and @instinct-vfx

File: src/rez/tests/test_shells.py [hunk 2]
```diff
@@ -331,8 +333,9 @@ def _print(value):
 
         # We are wrapping all variable outputs in quotes in order to make sure
         # our shell isn't interpreting our output as instructions when echoing
-        # it but this means we need to wrap our expected output as well.
-        expected_output = ['"{}"'.format(o) for o in expected_output]
+        # it but this means we need to wrap our expected output as well. Only
+        # exception is empty string, which is just passed through.
+        expected_output = ['"{}"'.format(o) if o else o for o in expected_output]
```
Comment (bfloch) [inline src/rez/tests/test_shells.py]: This was necessary since the interpretation of empty string in quotes would not match throughout the shells.
Not sure if this is correct but the data itself is correct (empty) and this is what I wanted to compare here.

**Comment by instinct-vfx** [comment]
I want to take a closer look at the code. While this does strip out the ECHO output it still leaves empty lines for me. Here is a comparison of 2.47.11, 2.47.13 and the current state of this MR.

While the empty lines are no show stopper they are still not needed and i am a bit confused by the difference. Why were the newlines introduced? And why does 2.47.11 not have the "you are now in a rez environment"?

**2.47.11**
image

**2.47.13**
image

**This MR**
image

**Comment by bfloch** [comment]
I addressed the message in a recent fix, I just matched it to the other shells.
https://github.com/nerdvegas/rez/commit/b277d26d8d76f28c0e05b0a60158b5d03009c96d

As to the newline:
`info('')` not meant to generate a newline. Else you would just leave it out.
I think this matches the other shells implementations.
So the MR is conforming to other shells and probably what we want.

**Comment by bfloch** [comment]
Here in cmd:
https://github.com/nerdvegas/rez/blame/master/src/rezplugins/shell/cmd.py#L165
Here in the UnixShells:
https://github.com/nerdvegas/rez/blob/master/src/rez/shells.py#L370
